### PR TITLE
feat: add automatic TenantId logging to Logger default properties

### LINKED ISSUE #2348: Feature request: Add TenantId to Logger default properties
### Use case

With the new Lambda Tenant Isolation feature, we have access to the TenantId from the LambdaContext object.
We should add the TenantId to the default properties in the Logging utility.

https://docs.aws.amazon.com/lambda/latest/dg/tenant-isolation.html

### Solution/User Experience

We need to add support for both logging backends, log4j2 and logback. The outcome should be that (without the user changing any code) the emitted JSON will contain `tenant_id`. 

```json
{
    "level": "Information",
    "message": "User object: Doe, John (42)",
    "timestamp": "2025-04-07 09:06:30.708",
    "service": "user-service",
    "coldStart": true,
    "name": "...",
    "user": {
        "firstName": "John",
        "lastName": "Doe",
        "age": 42
    },
    ...
    "tenantId": "cff02b3a-0e12-4be2-b3e0-758b49c4cd9b"
    ...
}
```

### Implementation suggestion

For Log4j2:
- Update JSON layouts in https://github.com/aws-powertools/powertools-lambda-java/tree/main/powertools-logging/powertools-logging-log4j/src/main/resources
- Create a new tenant id resolver in https://github.com/aws-powertools/powertools-lambda-java/blob/main/powertools-logging/powertools-logging-log4j/src/main/java/org/apache/logging/log4j/layout/template/json/resolver/PowertoolsResolver.java

For Logback:
- Update https://github.com/aws-powertools/powertools-lambda-java/blob/main/powertools-logging/powertools-logging-logback/src/main/java/software/amazon/lambda/powertools/logging/logback/LambdaJsonEncoder.java
- Update https://github.com/aws-powertools/powertools-lambda-java/blob/main/powertools-logging/powertools-logging-logback/src/main/java/software/amazon/lambda/powertools/logging/logback/LambdaEcsEncoder.java

### Alternative solutions

```Markdown

```

### Acknowledgment

- [x] This feature request meets [Powertools for AWS Lambda (Java) Tenets](https://docs.powertools.aws.dev/lambda/java/latest/#tenets)
- [x] Should this be considered in other Powertools for AWS Lambda languages? i.e. [Python](https://github.com/aws-powertools/powertools-lambda-python/), [TypeScript](https://github.com/aws-powertools/powertools-lambda-typescript/), and [.NET](https://github.com/aws-powertools/powertools-lambda-dotnet/)

### Future readers

Please react with 👍 and your use case to help us understand customer demand.

### PR DESCRIPTION
**Pull request by Pratapchandradeo**
## Description

Implements automatic `tenant_id` logging for AWS Lambda Tenant Isolation feature with null-safe serialization in Logback JSON encoder. Ensures `tenant_id` is only logged when it contains a valid value.

## Changes

✅ Add null-safety check to `LambdaJsonEncoder.serializePowertools()`
✅ Add `TENANT_ID_RESOLVER` in Log4j2 `PowertoolsResolver`
✅ Add `TENANT_ID` enum to `PowertoolsLoggedFields`
✅ Update `LambdaJsonLayout.json` and `LambdaEcsLayout.json`
✅ Add `tenant_id` serialization to `LambdaEcsEncoder`
✅ Add unit tests for null and empty `tenant_id` scenarios
✅ Fix test assertion and OS compatibility issues

## Implementation Details

- Skip `tenant_id` serialization when null or empty
- Follows pattern used for `sampling_rate`
- Supports both Log4j2 and Logback backends
- No user code changes required

## Testing

Tests run: 18, Failures: 0, Errors: 0, BUILD SUCCESS

Test coverage:
- Valid `tenant_id` is logged
- Null `tenant_id` is NOT logged
- Empty `tenant_id` is NOT logged
- Respects `includePowertoolsInfo` flag

## Related Issues

Closes #2348

## By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

- [x] I confirm that I can use, modify, copy, and redistribute this contribution under the terms of your choice

## Acknowledgment

- [x] This feature request meets [Powertools for AWS Lambda (Java) Tenets](https://docs.powertools.aws.dev/lambda/java/latest/#tenets)
- [x] I have read the [CONTRIBUTING](https://github.com/aws-powertools/powertools-lambda-java/blob/main/CONTRIBUTING.md) guidelines
- [x] Tests pass locally
- [x] No breaking changes

### CODE REVIEW

**Comment by phipag** [comment]
Hey @Pratapchandradeo thanks for opening this PR. Please check the PR template before we can proceed with the review: https://github.com/aws-powertools/powertools-lambda-java/blob/main/.github/PULL_REQUEST_TEMPLATE.md.

File: powertools-logging/src/main/java/software/amazon/lambda/powertools/logging/internal/PowertoolsLoggedFields.java [hunk 1]
```diff
@@ -55,6 +56,10 @@ public static Map<String, String> setValuesFromLambdaContext(Context context) {
         hashMap.put(FUNCTION_ARN.name, context.getInvokedFunctionArn());
         hashMap.put(FUNCTION_MEMORY_SIZE.name, String.valueOf(context.getMemoryLimitInMB()));
         hashMap.put(FUNCTION_REQUEST_ID.name, String.valueOf(context.getAwsRequestId()));
+        String tenantId = context.getTenantId();
+        if (tenantId != null && !tenantId.isEmpty()) {
```
Comment (phipag) [inline powertools-logging/src/main/java/software/amazon/lambda/powertools/logging/internal/PowertoolsLoggedFields.java:60]: Can we add a unit test for tenantId being `null` and `empty`?

File: powertools-logging/powertools-logging-logback/src/main/java/software/amazon/lambda/powertools/logging/logback/LambdaEcsEncoder.java [hunk 2]
```diff
@@ -169,6 +171,12 @@ private void serializeFunctionInfo(JsonSerializer serializer, String arn, Map<St
                 serializer.writeRaw(',');
                 serializer.writeStringField(CORRELATION_ID_ATTR_NAME, correlationId);
             }
+            
+            String tenantId = mdcPropertyMap.get(TENANT_ID.getName());
+            if (tenantId != null) {
```
Comment (phipag) [inline powertools-logging/powertools-logging-logback/src/main/java/software/amazon/lambda/powertools/logging/logback/LambdaEcsEncoder.java]: I think we should check for both null and empty string similar to the `PowertoolsLoggedFields`.

File: powertools-logging/powertools-logging-log4j/src/main/java/org/apache/logging/log4j/layout/template/json/resolver/PowertoolsResolver.java [hunk 3]
```diff
@@ -186,6 +187,23 @@ public void resolve(LogEvent logEvent, JsonWriter jsonWriter) {
         }
     };
 
+    private static final EventResolver TENANT_ID_RESOLVER = new EventResolver() {
+        @Override
+        public boolean isResolvable(LogEvent logEvent) {
+            final String tenantId =
+                    logEvent.getContextData().getValue(PowertoolsLoggedFields.TENANT_ID.getName());
+            return null != tenantId;
```
Comment (phipag) [inline powertools-logging/powertools-logging-log4j/src/main/java/org/apache/logging/log4j/layout/template/json/resolver/PowertoolsResolver.java]: I think we should check for both null and empty string similar to the `PowertoolsLoggedFields`.

**Review by phipag — CHANGES_REQUESTED** [review]
Also some first comments on the code :-)

**Comment by phipag** [comment]
@Pratapchandradeo thanks for the update. Please check my comment about the PR description format. The acknowledgement section is incorrect.

**Comment by Pratapchandradeo** [comment]
Hi! @phipag,  CI is failing because adding `tenant_id` changes the MDC size and ECS output (`tenant.id`), so `LambdaLoggingAspectTest` and `LambdaEcsEncoderTest` assertions need updating. Should I update those tests to include the new field, or would you prefer a different approach?

**Comment by phipag** [comment]
Yes please update these tests @Pratapchandradeo and make sure the build for all modules passes locally on your computer as well.

**Comment by Pratapchandradeo** [comment]
Done ✅

- Updated `LambdaLoggingAspectTest` expectations to account for the new default `tenant_id` field
- Updated `LambdaEcsEncoderTest.shouldLogInEcsFormat` to include `tenant.id`
- Made the ECS exception stack assertion tolerant of both LF and CRLF

Verified locally:
- `mvn clean test -pl powertools-logging -Dtest=LambdaLoggingAspectTest` ✅
- `mvn clean test -pl powertools-logging/powertools-logging-logback -Dtest=LambdaEcsEncoderTest,LambdaJsonEncoderTest` ✅

**Comment by Pratapchandradeo** [comment]
Hi! @phipag,

I’m seeing the build still fail because `PowerToolsResolverFactoryTest` uses a strict long substring assertion, and the new `tenant_id` / `tenant.id` field breaks the expected ordering.

Would you recommend I keep splitting these into smaller `contains(...)` checks, or is there a preferred assertion style for these Log4j tests?

**Comment by phipag** [comment]
Hey @Pratapchandradeo, it is your call. Please make a decision on the testing strategy and once it is working I am happy to review it. 

Did you verify before updating the PR that the tests pass locally? You only ran tests for the changed files. My recommendation is to run tests for the whole logging module and then for the whole project to make sure that it actually works.

**Review by phipag — APPROVED** [review]
Thanks for working on this @Pratapchandradeo. This looks good to me now. Merging your PR.

Congratulations on your first contribution this project 🥳
